File: citypicker/src/main/java/com/liabit/citypicker/adapter/decoration/DividerItemDecoration.java
package com.liabit.citypicker.adapter.decoration;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Rect;
import android.util.TypedValue;
import android.view.View;

import com.liabit.citypicker.R;

import androidx.recyclerview.widget.RecyclerView;

public class DividerItemDecoration extends RecyclerView.ItemDecoration {
    private float dividerHeight;
    private Paint mPaint;

    public DividerItemDecoration(Context themeContext) {
        mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);

        TypedValue typedValue = new TypedValue();
        themeContext.getTheme().resolveAttribute(R.attr.cpSectionBackground, typedValue, true);
        mPaint.setColor(themeContext.getResources().getColor(typedValue.resourceId));
        dividerHeight = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 0.5f, themeContext.getResources().getDisplayMetrics());
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        outRect.bottom = (int) dividerHeight;
    }

    @Override
    public void onDraw(Canvas c, RecyclerView parent, RecyclerView.State state) {
        int childCount = parent.getChildCount();
        int left = parent.getPaddingLeft();
        int right = parent.getWidth() - parent.getPaddingRight();

        for (int i = 0; i < childCount - 1; i++) {
            View view = parent.getChildAt(i);
            float top = view.getBottom();
            float bottom = view.getBottom() + dividerHeight;
            c.drawRect(left, top, right, bottom, mPaint);
        }
    }
}

File: citypicker/src/main/kotlin/com/liabit/listpicker/decoration/DividerItemDecoration.kt
package com.liabit.listpicker.decoration

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Rect
import android.util.TypedValue
import android.view.View
import androidx.core.content.res.ResourcesCompat
import androidx.recyclerview.widget.RecyclerView

class DividerItemDecoration(context: Context) : RecyclerView.ItemDecoration() {
    private val mDividerHeight: Float
    private val mPaint = Paint(Paint.ANTI_ALIAS_FLAG)

    init {
        val typedValue = TypedValue()
        context.theme.resolveAttribute(com.liabit.citypicker.R.attr.cpSectionBackground, typedValue, true)
        mPaint.color = ResourcesCompat.getColor(context.resources, typedValue.resourceId, context.theme)
        mDividerHeight = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 0.5f, context.resources.displayMetrics)
    }

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        outRect.bottom = mDividerHeight.toInt()
    }

    override fun onDraw(canvas: Canvas, parent: RecyclerView, state: RecyclerView.State) {
        val childCount: Int = parent.childCount
        val left: Int = parent.paddingLeft
        val right: Int = parent.width - parent.paddingRight
        for (i in 0 until childCount - 1) {
            val view: View = parent.getChildAt(i)
            val top = view.bottom
            val bottom = view.bottom + mDividerHeight
            canvas.drawRect(left.toFloat(), top.toFloat(), right.toFloat(), bottom, mPaint)
        }
    }


}
